src/Data/Argonaut/Encode/Implementation.purs

```diff
@@ -0,0 +1,122 @@
+module Data.Argonaut.Encode.Implementation where
+
+import Prelude
+
+import Data.Argonaut.Core (Json, fromArray, fromBoolean, fromNumber, fromObject, fromString, jsonNull, caseJsonObject, jsonSingletonObject)
+import Data.Array as Arr
+import Data.Array.NonEmpty (NonEmptyArray)
+import Data.Array.NonEmpty as NEA
+import Data.Either (Either, either)
+import Data.Identity (Identity(..))
+import Data.Int (toNumber)
+import Data.List (List(..), (:), toUnfoldable)
+import Data.List.NonEmpty as NEL
+import Data.List.Types (NonEmptyList)
+import Data.Map as M
+import Data.Maybe (Maybe(..))
+import Data.NonEmpty (NonEmpty(..))
+import Data.Set as S
+import Data.String (CodePoint)
+import Data.String.CodePoints as CP
+import Data.String.CodeUnits as CU
+import Data.Tuple (Tuple(..))
+import Foreign.Object as FO
+
+type Encoder a = a -> Json
+
+encodeIdentity :: forall a . Encoder a -> Encoder (Identity a)
+encodeIdentity encoder (Identity a) = encoder a
+
+encodeJsonMaybe :: forall a . Encoder a -> Encoder (Maybe a)
+encodeJsonMaybe encoder = case _ of
+    Nothing -> jsonNull
+    Just a -> encoder a
+
+encodeJsonTuple :: forall a b . Encoder a -> Encoder b -> Encoder (Tuple a b)
+encodeJsonTuple encoderA encoderB (Tuple a b) = fromArray [encoderA a, encoderB b]
+
+encodeJsonEither :: forall a b . Encoder a -> Encoder b -> Encoder (Either a b)
+encodeJsonEither encoderA encoderB = either (obj encoderA "Left") (obj encoderB "Right")
+    where
+    obj :: forall c. Encoder c -> String -> c -> Json
+    obj encoder tag x =
+      fromObject
+        $ FO.fromFoldable
+        $ Tuple "tag" (fromString tag) : Tuple "value" (encoder x) : Nil
+
+encodeJsonUnit :: Encoder Unit
+encodeJsonUnit = const jsonNull
+
+encodeJsonJBoolean :: Encoder Boolean
```

Comment:
I recognize that the instance was called `JBoolean`, but I think we should drop the `J` prefix and just use `Boolean` for this function, as well as for `Number`, `String`, and so on.

Reply:
Ok

Reply:
done